### PR TITLE
feat: stan reliable auto reconnection

### PR DESCRIPTION
**Pull request by whynowy**
Signed-off-by: Derek Wang <whynowy@gmail.com>

Closes: #95

### CODE REVIEW

File: runner/sidecar/sinks.go [hunk 1]
```diff
@@ -134,38 +134,55 @@ func connectKafkaSink(x *dfv1.Kafka, sinkName string) (func(msg []byte) error, e
...
-		return sc.Close()
+		if conn != nil && !conn.IsClosed() {
+			return conn.Close()
+		}
+		return nil
 	})
-	f := func(msg []byte) error { return sc.Publish(x.Subject, msg) }
+
+	go func() {
+		logger.Info("starting stan auto reconnection daemon", "sink", sinkName)
```
Comment (alexec) [inline runner/sidecar/sinks.go:163]: can you add `runtime.HandleCrash()` so we at least get a log of any panic

File: runner/sidecar/sinks.go [hunk 2]
```diff
@@ -134,38 +134,55 @@ func connectKafkaSink(x *dfv1.Kafka, sinkName string) (func(msg []byte) error, e
...
+		ticker := time.NewTicker(5 * time.Second)
```
Comment (alexec) [inline runner/sidecar/sinks.go:164]: minor - I prefer loop with time.Sleep, but potatoes/tomatoes

File: runner/sidecar/sources.go [hunk 3]
```diff
@@ -213,63 +213,93 @@ func registerKafkaSetPendingHook(x *dfv1.Kafka, sourceName string, client sarama
...
 			return sub.Close()
 		})
-
 		if leadReplica() {
 			registerSTANSetPendingHook(sourceName, x, queueName)
 		}
 	}
+
+	go func() {
+		logger.Info("starting stan auto reconnection daemon", "source", sourceName)
```
Comment (alexec) [inline runner/sidecar/sources.go:274]: runtime.HandleCrash

**Review by alexec — CHANGES_REQUESTED** [review]
this is a complex PR - I wonder if it might be worthwhile abstracting stan (and maybe other sources and sinks) behind an interface that adds re-connection etc - but not today - some minor comments

File: runner/sidecar/sources.go (hunk 3 above)
Comment (whynowy) [reply runner/sidecar/sources.go:276]: done.

File: runner/sidecar/sinks.go (hunk 2 above)
Comment (whynowy) [reply runner/sidecar/sinks.go]: done.

File: runner/sidecar/sinks.go (hunk 1 above)
Comment (whynowy) [reply runner/sidecar/sinks.go:165]: done.

**Review by alexec — APPROVED** [review]

**Pull request merged**